### ISSUE TITLE
TVos support?

### ISSUE THREAD

**Issue by onyx4** [ndfred/iperf-ios#33]
Would it be possible to make the app available for Apple TV?   I have an Apple TV connected using Ethernet over Power and would like to test the speed using iperf3.    That would be also useful validating the speed of the wifi/ethernet on Apple TV with a local network source.  Currently only speedtest app is available on Apple TV.

**Comment by aptonline** [comment]
I too would be interested. Not sure how easy it would be easy to compile for tvOS.

**Comment by ndfred** [comment]
That would be nice for sure, a few differences that would make it a challenge to implement:

* text input to pick a server would be an issue, best would be to scan the network for iPerf servers but that's not implemented yet
* Apple TV has WiFi and Ethernet, potentially at the same time
* Apple TV is much more challenging to deploy / debug on
* the UI would have to be completely redone
* one extra interface to maintain, and which would probably not get as much attention as the other more popular variants

Realistically I don't think I'll have time to work on an Apple TV version soon. I would like to rewrite the app in Swift / SwiftUI and have it be available on macOS as well, at which point the gap to a tvOS version might be narrower.

In the meantime, I wonder if you could get a [local SpeedTest server](https://help.speedtest.net/hc/en-us/articles/360039164793-How-can-I-host-a-Speedtest-server-) set up? Seems unlikely to be possible but you never know.

**Comment by aptonline** [comment]
@ndfred Thanks for your reply.  Great points you raise there.  I may fork your repository and have a play :).

With regard to local SpeedTest server I didn't realise SpeedTest could be used for local networks or is that setting top a server for external use?

**Comment by aptonline** [comment]
Out of interest I found this post of the dev forums, basically looks like you can add tvOS support by adding an additional target and then design the UI in storyboard.  The code should work as is:

https://developer.apple.com/forums/thread/95890

**Comment by ndfred** [comment]
@aptonline: if you are looking for a reason to build a tvOS app, by all means fork the project and go for it! The backend is separated enough from the frontend that it shouldn't be too much of a hassle. I didn't use Storyboards or NIBs so you won't have IBOutlets and all there for you, but you can always write your own view controller to add these.

The hold up for me is the time it would take to make the interface good / reliable enough that I would feel comfortable publishing it to the App Store.

As an aside: are you using Ethernet for your Apple TV? If you care a lot about its network performance you probably already are, which means you are pretty much guaranteed gigabit speeds. I use [SmokePing](https://oss.oetiker.ch/smokeping/) to monitor the connection quality, and ran Speedtest at some point to make sure I am seeing Gigabit speeds, and feel confident that the Apple TV is set up correctly. WiFi is a different matter, which is where I have found iPerf to be helpful.

**Comment by aptonline** [comment]
Also with regard to Ethernet and Wifi, if an Ethernet cable is connected this take precedent over wifi there doesn't seem to be a way to use both at once, that's how it works for me at least.

**Comment by aptonline** [comment]
@ndfred Yes I am using Ethernet but other Powerline (TP-Link plugs) so want to make sure I get good speeds but this in reliant on electrical infrastructure.

**Comment by onyx4** [comment]
Same here, I use Ethernet but over powerline at the TV location.  So I wanted to compare with a local iPerf server if my WiFi 5Ghz or Powerline was faster. 

Thanks,
Christian 

On Jun 29, 2020, at 7:33 AM, aptonline <notifications@github.com> wrote:

﻿
@ndfred Yes I am using Ethernet but other Powerline (TP-Link plugs) so want to make sure I get good speeds but this in reliant on electrical infrastructure.

**Comment by ndfred** [comment]
That makes sense, I've always found a way to run Ethernet after having a terrible experience with Powerline, but that doesn't work everywhere. About network interfaces, I am indeed not seeing my Apple TV connected over WiFi when it is wired so that might not be an issue. Another interesting thought is that an Apple TV could become a wired iPerf server that WiFi clients could connect to (if it isn't Powerline).

**Comment by aptonline** [comment]
Unfortunately I've not had the opportunity to look at this :( still very interested though as I think it would fill a need for local network speed tests as there is nothing like it on the App Store for AppleTV.

**Comment by jgstew** [comment]
The use case I have and went looking for iperf on AppleTV is that I want to test my local WiFi speed to the Apple TV that is connected via Ethernet or test WiFi to WiFi. I'm in an Airbnb at the moment and it would be helpful for me to compare both options for viability of airplay mirroring from my laptop to the AppleTV and just in general.

**Comment by pcmike** [comment]
Just dropping a comment here in 2024 to say I would love to see iperf3 on tvOS. Always nice to test “ethernet connectivity” to an Apple TV that in reality is fed by other non traditional means (MoCA in my case). 👍🏻
